clang/lib/Analysis/UnsafeBufferUsage.cpp

```diff
@@ -2495,10 +2471,100 @@ static FixItList fixVariableWithSpan(const VarDecl *VD,
   return fixLocalVarDeclWithSpan(VD, Ctx, getUserFillPlaceHolder(), Handler);
 }
 
+static FixItList fixVarDeclWithArray(const VarDecl *D, const ASTContext &Ctx,
+                                     UnsafeBufferUsageHandler &Handler) {
+  FixItList FixIts{};
+
+  if (auto CAT = dyn_cast<clang::ConstantArrayType>(D->getType())) {
+    const QualType &ArrayEltT = CAT->getElementType();
+    assert(!ArrayEltT.isNull() && "Trying to fix a non-array type variable!");
+    // FIXME: support multi-dimensional arrays
+    if (isa<clang::ConstantArrayType>(ArrayEltT))
```

Comment:
is it possible that the element type is an array type but not a constant array type?

Reply:
You're right, for example it's possible to mix CSA with C variable length arrays.
https://clang.llvm.org/doxygen/classclang_1_1VariableArrayType.html

```
void foo(unsigned u) {
    int foo[5][u];
}

```
Let me fix this and add a testcase.

Reply:
...and on top of that the `if` should check for canonical type to not be fooled by typedefs & co.